### PR TITLE
[IOTDB-3316] Add 'SHOW CLUSTER' implementation for IoTDB cluster mode

### PR DESCRIPTION
**Pull request by Beyyes**
## Description

This pr adds the implementation for 'SHOW CLUSTER' command.  It displays the 'NodeId、Role、Status、Host、Ports' informations of current cluster.

See more details in JIRA: https://issues.apache.org/jira/browse/IOTDB-3316

The output result of 'SHOW CLUSTER' is as below .
image

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/mpp/plan/execution/config/ShowClusterTask.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    }
+
+    // build TSBlock
+    TsBlockBuilder builder =
+        new TsBlockBuilder(HeaderConstant.showClusterHeader.getRespDataTypes());
+    int configNodeId = 0;
+    for (TConfigNodeLocation nodeLocation : clusterNodeInfos.getConfigNodeList()) {
+      builder.getTimeColumnBuilder().writeLong(0L);
+      builder.getColumnBuilder(0).writeInt(configNodeId++);
+      builder.getColumnBuilder(1).writeBinary(new Binary("ConfigNode"));
```
Comment (xingtanzjr) [inline server/src/main/java/org/apache/iotdb/db/mpp/plan/execution/config/ShowClusterTask.java:83]: Suggest to use constant defined in IoTDBConstant, or use ENUM here

File: server/src/main/java/org/apache/iotdb/db/mpp/plan/execution/config/ShowClusterTask.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+    if (config.isClusterMode()) {
+      try (ConfigNodeClient client = clientManager.borrowClient(ConfigNodeInfo.partitionRegionId)) {
+        clusterNodeInfos = client.getAllClusterNodeInfos();
+      } catch (TException | IOException e) {
+        LOGGER.error("Failed to connect to config node.");
+        future.setException(e);
+      }
+    } else {
+      future.setException(new TException("SHOW CLUSTER not support local mode"));
```
Comment (xingtanzjr) [inline server/src/main/java/org/apache/iotdb/db/mpp/plan/execution/config/ShowClusterTask.java:73]: `does not`

File: server/src/main/java/org/apache/iotdb/db/mpp/plan/execution/config/ShowClusterTask.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    for (TDataNodeLocation nodeLocation : clusterNodeInfos.getDataNodeList()) {
+      builder.getTimeColumnBuilder().writeLong(0L);
+      builder.getColumnBuilder(0).writeInt(nodeLocation.getDataNodeId());
+      builder.getColumnBuilder(1).writeBinary(new Binary("DataNode"));
+      builder.getColumnBuilder(2).writeBinary(new Binary("Running"));
+      builder
+          .getColumnBuilder(3)
+          .writeBinary(new Binary(nodeLocation.getInternalEndPoint().getIp()));
+      builder.getColumnBuilder(4).writeInt(nodeLocation.getInternalEndPoint().getPort());
+      builder.declarePosition();
```
Comment (xingtanzjr) [inline server/src/main/java/org/apache/iotdb/db/mpp/plan/execution/config/ShowClusterTask.java:100]: Line `92-100` is similar with line `81-89`, suggest to define a function to process it.

File: server/src/main/java/org/apache/iotdb/db/mpp/plan/execution/config/ShowClusterTask.java [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+import java.io.IOException;
+
+public class ShowClusterTask implements IConfigTask {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ShowClusterTask.class);
+
+  private static final IoTDBConfig config = IoTDBDescriptor.getInstance().getConfig();
+
+  private ShowClusterStatement showClusterStatement;
```
[the remark this answers is not in the document]
Comment (Beyyes) [reply server/src/main/java/org/apache/iotdb/db/mpp/plan/execution/config/ShowClusterTask.java]: remove it

File: server/src/main/java/org/apache/iotdb/db/mpp/plan/execution/config/ShowClusterTask.java (hunk 1 above)
Comment (Beyyes) [reply server/src/main/java/org/apache/iotdb/db/mpp/plan/execution/config/ShowClusterTask.java]: used constants defined in IoTDBConstant

File: server/src/main/java/org/apache/iotdb/db/mpp/plan/execution/config/ShowClusterTask.java (hunk 3 above)
Comment (Beyyes) [reply server/src/main/java/org/apache/iotdb/db/mpp/plan/execution/config/ShowClusterTask.java]: Line 92-100 and line 81-89 are extracted in a function

**Review by xingtanzjr — APPROVED** [review]

**Pull request merged**